src/main/java/io/phdata/streamliner/schemadefiner/ConfigBuilder/DiffGenerator.java

```diff
@@ -0,0 +1,186 @@
+                                            .get(0);
+                                    ColumnDiff colDiff =
+                                        new ColumnDiff(null, currColDef, false, true, false);
+                                    columnDiffs.add(colDiff);
+                                  });
+
+                          commonColumns.stream()
+                              .forEach(
+                                  commonCol -> {
+                                    ColumnDefinition prevColDef =
+                                        prevColDefList.stream()
+                                            .filter(
+                                                colDef -> colDef.getSourceName().equals(commonCol))
+                                            .collect(Collectors.toList())
+                                            .get(0);
+                                    ColumnDefinition currColDef =
+                                        currColDefList.stream()
+                                            .filter(
+                                                colDef -> colDef.getSourceName().equals(commonCol))
+                                            .collect(Collectors.toList())
+                                            .get(0);
+                                    if (!prevColDef.equals(currColDef)) {
+                                      updatedColumns.add(prevColDef.getSourceName());
+                                      ColumnDiff colDiff =
+                                          new ColumnDiff(
+                                              prevColDef, currColDef, false, false, true);
+                                      columnDiffs.add(colDiff);
+                                    }
+                                  });
+
+                          log.debug(
+                              "Table: {}, DeletedColumns: {}, AddedColumns: {}, UpdatedColumns: {}",
+                              prevTableDef.getSourceName(),
+                              deletedColumns,
+                              addedColumns,
+                              updatedColumns);
+
+                          if (!columnDiffs.isEmpty()) {
+                            tableDiff =
+                                new TableDiff(
+                                    currTableDef.getType(),
+                                    currTableDef.getDestinationName(),
+                                    columnDiffs,
+                                    true);
+                            tableDiffList.add(tableDiff);
+                          }
+                        }
+                      });
+            });
+    return tableDiffList;
+  }
+
+  private static List<String> getCommonCol(List<String> list1, List<String> list2) {
+    List<String> tempList1 = new ArrayList<>(list1);
+    List<String> tempList2 = new ArrayList<>(list2);
+    tempList1.retainAll(tempList2);
+    return tempList1;
+  }
```

Comment:
retainAll on two lists is o(N). In addition for large deployments we'll be doing this thousands of times. I could see this being a performance issue.

Could you update the type in this class from List to LinkedHashSet. LinkedHashSet provides O(n) lookup in the normal case and also provides insertion based ordering like a List. Thus I think for this use case we probably want to use LinkedHashSet everywhere.